contrib/externalMatchfinder/main.c

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) Yann Collet, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under both the BSD-style license (found in the
+ * LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ * in the COPYING file in the root directory of this source tree).
+ * You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define ZSTD_STATIC_LINKING_ONLY
+#include "zstd.h"
+#include "zstd_errors.h"
+#include "matchfinder.h" // simpleExternalMatchFinder
+
+int main(int argc, char *argv[]) {
+    size_t res;
+
+    if (argc != 2) {
+        printf("Usage: exampleMatchfinder <file>\n");
+        return 1;
+    }
+
+    ZSTD_CCtx* zc = ZSTD_createCCtx();
+
+    int simpleExternalMatchState = 0xdeadbeef;
+
+    // Here is the crucial bit of code!
+    ZSTD_refExternalMatchFinder(
+        zc,
+        &simpleExternalMatchState,
+        simpleExternalMatchFinder
+    );
+
+    res = ZSTD_CCtx_setParameter(zc, ZSTD_c_enableMatchFinderFallback, 1);
```

Comment:
minor : coding style

`res` is only used immediately after, to test its value, and output an error if need be.

It's unrelated to the later set `res` at decompression time.

Prefer to limit variable's lifetime,
this makes it more clear that the value set in `res` is not meant to be-reused later on.
This can avoid future maintenance issues,
such as forgetting to set `res` later (like at decompression time), but still test it, 
misleadingly believing it represents the result of the most recent action.

The easiest way to reduce `res` lifetime is to create a new `{ }` block scope.

Marginally more complex, you may encapsulate the whole "test, write error and return" logic in a macro,
higher investment but this may even make the code's logic clearer.